app/controllers/mixins/ems_common_angular.rb

```diff
@@ -266,6 +266,20 @@ def ems_form_fields
                         :ssh_keypair_auth_status       => ssh_keypair_auth_status.nil? ? true : ssh_keypair_auth_status
       } if controller_name == "ems_infra"
 
+      render :json => {
+        :name                => @ems.name,
+        :emstype             => @ems.emstype,
+        :zone                => zone,
+        :provider_id         => @ems.provider_id ? @ems.provider_id : "",
+        :hostname            => @ems.hostname,
+        :default_hostname    => @ems.connection_configurations.default.endpoint.hostname,
+        :default_api_port    => @ems.connection_configurations.default.endpoint.port,
+        :provider_region     => @ems.provider_region,
+        :default_userid      => @ems.authentication_userid ? @ems.authentication_userid : "",
+        :ems_controller      => controller_name,
+        :default_auth_status => default_auth_status,
+      } if controller_name == "ems_physical_infra"
```

Comment:
> Well if we keep saying that and postponing cleanups then we are doomed.

This is a fine sentiment, but this `ems_common` has been bad for a very *very* long time.  Pinning this level of cleanup on this PR is a mistake, imo.

Refactoring `ems_common` to be what it really should be (common code used by multiple controllers) is a monumental effort.  Best suited for a separate dedicated PR that can be reviewed in that specific context.